diff --git a/minion.py b/minion.py
index d7f5653..a2e3817 100644
--- a/minion.py
+++ b/minion.py
@@ -1916,6 +1916,7 @@ class Minion(MinionBase):
                 return_data = minion_instance._execute_job_function(
                     function_name, function_args, executors, opts, data
                 )
+                log.warning("{} args: {}".format(function_name, function_args), exc_info_on_loglevel=logging.DEBUG)
 
                 if isinstance(return_data, types.GeneratorType):
                     ind = 0
